README.md

```diff
@@ -97,60 +97,76 @@ let encodedTrifleSignedDataProto = try trifleSignedData.serialize()
 ```kotlin
 // Check if a key already exists.
 // If no key exists, generate a public key pair
-var keyHandle =  TrifleApi.generateKeyHandle(alias: abc)
+var keyHandle: KeyHandle =  TrifleApi.generateKeyHandle("abc")
                 
-// Storing keys. Keys are codable.
-val encoder = Gson()
-let jsonKeyHandle = try encoder.toJson(keyHandle)
+// Storing keys. Key handles are serializable.
+val encoded: ByteArray = keyHandle.serialize()
 
 // Load the key from storage when we need to use it
+val decoded: KeyHandle = KeyHandle.deserialize(encoded)
 
 // Check the validity of loaded key
-// TBD trifleApi.isValid(keyHandle: keyHandle)
+val valid: Boolean = trifleApi.isValid(keyHandle)
 
 // Destroy key that is no longer in use or is invalid
-// TBD trifleApi.delete(keyHandle: keyHandle)
+trifleApi.delete(keyHandle)
         
 // Check if loaded key already has a cert. If yes, skip to checking for cert validity
 // Else if key does not have a cert OR if a new cert must be generated (eg because of existing
 // cert is already expired, or app needs to re-attest, app is re-installed, app is restored
 // from backup, ... etc)
 
 // Create cert request
-val certReq = TrifleApi.generateMobileCertificateRequest(entity = entity, keyHandle = keyHandle)
+val certReq: CertificateRequest = TrifleApi.generateMobileCertificateRequest(entity, keyHandle)
 
-// Serialize to proto to be sent over wire
+// Serialize as an opaque ByteArray
+val encoded: ByteArray = certReq.serialize()
 
-// Send certificate request to Certificate Authority endpoint. Response will be [Data]
-// Iterate over each Data to convert to TrifleCertificate
+// Send certificate request to Certificate Authority endpoint. Response will be List<ByteArray>
+val response: List<ByteArray>
```

Comment:
we actually get `ByteString`
![image](https://github.com/cashapp/trifle/assets/20655571/e05e645a-2e6c-4896-8d07-6032a3a014f0)

Reply:
Our endpoint uses Wire so bytes are expressed as ByteString. However, our serialize methods return ByteArrays - so typically it should have been over a list of ByteArrays. This happen so to be an open source library so we'r trying not to tie it to what our infrastructure is